### ISSUE TITLE
[1.18.2] Flywheel Rendering Crash - Incompatibility with Rubidium and Oculus.

### ISSUE THREAD

**Issue by MokahTGS** [Engine-Room/Flywheel#136]
### Describe the Bug

Crash when exploring.  Possible Incompatibility with Rubidium and Oculus.

### Reproduction Steps

1.Have mods installed.
2.Explore world
3.Crash
...

### Expected Result

Not crashing

### Crash Report or Log

https://pastebin.com/JDTKn1nn

### Operating System

Win 10

### What is your GPU?

sure

### Mod Version

0.7.0

### Minecraft Version

1.18

### Loader Version

40.1.48

### Other Mods

TNP 5 Modpack

### Additional Context

none

**Comment by botafi** [comment]
Have the same crash and can reproduce. Crashes with the same logs when it tries to render the water wheel that was placed so the world is unplayable
Mods: Flywheel 0.6.2, Create 0.4.1, Oculus 1.2.5, Rubidium 0.5.2a

**Comment by f0kx** [comment]
hey try /flywheel backend off ingame and see if it helps...its only a workaround because you will see some stuff will break too exspacally with some create stuff rendering wise

**Comment by UdoMorgenstern** [comment]
maybe the fix/workaround can be the same as used for optifine shaders. 

As the crash only occurs when the flywheel backend is active and oculus shaders have been enabled
So just disable flywheel backend as soon as shaders are activated and vise versa

Quote from backend/Backend.java checking for optifine shaders as referece

```
private static BackendType chooseEngine() {
	BackendType preferredChoice = FlwConfig.get()
			.getBackendType();

	boolean usingShaders = OptifineHandler.isUsingShaders();
	boolean canUseEngine = switch (preferredChoice) {
		case OFF -> true;
		case BATCHING -> !usingShaders;
		case INSTANCING -> !usingShaders && GlCompat.getInstance().instancedArraysSupported();
	};

	return canUseEngine ? preferredChoice : BackendType.OFF;
}
```

**Comment by MokahTGS** [comment]
So the "fix" is to either not have shaders or not have create blocks render.  Can't we have an actual fix where we can have both?

**Comment by shikyo13** [comment]
I can reproduce the same crash, with the same mods. But when trying the provided workaround, /flywheel backend off I'm unable to join a server due to another error. Re-enabling backend but disabling Rubidium and Oculus allows me join. 

![image](https://user-images.githubusercontent.com/10563401/176248681-f0470cdf-9d39-4a1e-8bdd-60f8028e5c19.png)


Original Crash -

Minecraft Version
1.18.2

Flywheel Version
flywheel-forge-1.18-0.6.2

Iris Version
oculus-mc1.18.2-1.2.5

Sodium Version
rubidium-0.5.2a

Operating System
Win 10

What is your GPU?
sure

Java Version
Java 17

[crash-2022-06-28_11.36.49-client.txt](https://github.com/Jozufozu/Flywheel/files/9003582/crash-2022-06-28_11.36.49-client.txt)

**Comment by Brickmaster1** [comment]
>maybe the fix/workaround can be the same as used for optifine shaders.

>As the crash only occurs when the flywheel backend is active and oculus shaders have been enabled
>So just disable flywheel backend as soon as shaders are activated and vise versa

I have been thinking of this exact same fix for weeks bc it really bothers me that all my create stuff looks bugged with shaders. He needs to implement oculus getting detected and tweaked the same way optifine does. He says that the optimizations get disabled with optifine on and he needs to do the same with rubidium + oculus. Btw you should create a new thread with the suggestio to bring it to his attention.

**Comment by Jozufozu** [comment]
Fixed in 0.6.3+
